src/UI/Art/SpriteArtist/PlayerSetupArtist.java

```diff
@@ -63,6 +63,8 @@ public static void draw(Graphics g, MapInfo mapInfo, PlayerSetupController contr
         PlayerSetupAiArtist.draw(g, subMenu, control.getPlayerInfo(control.getHighlightedPlayer()).getCurrentColor());
       }
     }
+    // Start preloading infantry sprites in the background so the ColorFaction screen doesn't freeze on first entry.
+    PlayerSetupColorFactionArtist.preloadOneInfantrySprite();
```

Comment:
I feel like this is unorthodox, but it works.

Reply:
Yeah. But this way we get the same effect, and we don't have to mess with actual threading.